src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -900,6 +910,17 @@ class OpenShiftDSL implements Serializable {
             }
         }
 
+        public boolean exists() throws AbortException {
+            return names().size() > 0
```

Comment:
I hadn't previously picked up on this "static objectList based selector" notion from the code or associated comments, nor am I sure I'm fully gathering it from you explanation here.  Could you elaborate (either in this comment thread or face to face if you have a sec) ?

That said, the null checks for `objectList` and this snippet from the selector constructor did catch my eye previously:

```
            if ( qualifier != null ) {
                if ( qualifier instanceof Map ) {
                    this.labels = new HashMap((Map)qualifier);
                } else {
                    // Otherwise, the qualifier is a name that is paired with the name
                    objectList = Arrays.asList( kind + "/" + qualifier.toString() );
                    kind = null;
                }
            }

```

and was curious if that affected the previous suggested solution.

I'm thinking that is somewhat involved here then with what you found.  But I'd still like to get a better grasp of your terminology, etc.

All that said, I'll push up a new version with these changes later today.

Thanks.